### PR TITLE
TM-4306, add Vice/Vacancy to AIs

### PR DESCRIPTION
**Pull request by M00staff**
Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/813)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/376)

[Ticket](https://metaphase.atlassian.net/browse/TM-4306)

Show Vice in the Panel Meeting Page
image

But not in the Agenda Item History Page (since the vice will always be the CURRENT incumbent of that position)
image

And regardless - it will render in the Maintenance Page (as read only)
image

not entirely sure why we can edit old panel meetings?

### CODE REVIEW

**Comment by M00staff** [comment]
> Let's keep Vice in the same position on the different views

all set!

File: src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx [hunk 1]
```diff
@@ -115,6 +124,16 @@ const AgendaItemLegs = props => {
     },
   ];
 
+  if (isPanelMeetingView) { // vice/vacancy info only shows for panel view, and in AIM
```
Comment (pveland621) [inline src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx:127]: do we want to remove comments?

File: src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx [hunk 2]
```diff
@@ -113,7 +113,7 @@ const AgendaItemRow = props => {
               </div>
             </div>
           }
-          <AgendaItemLegs legs={agenda.legs} />
+          <AgendaItemLegs legs={agenda.legs} isPanelMeetingView={isPanelMeetingView} />
```
Comment (pveland621) [inline src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx:116]: can we shorten this to just `isPanelMeetingView`?

File: src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx [hunk 3]
```diff
@@ -25,6 +26,14 @@ const AgendaItemLegs = props => {
   const formatLang = (langArr = []) => langArr.map(lang => (
     `${lang.code} ${lang.spoken_proficiency}/${lang.reading_proficiency}`
   )).join(', ');
+  const formatVice = (viceObj) => {
```
Comment (pveland621) [inline src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx:29]: very nice

**Review by pveland621 — COMMENTED** [review]
code looks good to me, finishing up functional testing

File: src/sass/_agendaItemRow.scss [hunk 4]
```diff
@@ -76,7 +76,6 @@
         th {
           border: none;
           padding: .5rem 1rem;
-          max-width: 80px;
```
Comment (elizabeth-jimenez) [inline src/sass/_agendaItemRow.scss:79]: What was the motivation for this?

Comment (elizabeth-jimenez) [reply src/sass/_agendaItemRow.scss:79]: That makes sense - I think we may have implemented that before we added a scroll to the AILegs portion. Nice catch

File: src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx (hunk 2 above)
Comment (M00staff) [reply src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx:116]: negative - passing props down that way would defaults to `true`

File: src/Components/Agenda/AgendaLeg/AgendaLeg.jsx [hunk 5]
```diff
@@ -245,6 +245,15 @@ const AgendaLeg = props => {
     </div>
   );
 
+  const getVice = (viceObj) => {
+    const vice = viceObj?.emp_full_name;
+    const vacancy = viceObj?.asgd_etd_ted_date && formatDate(viceObj.asgd_etd_ted_date, 'MM/YY');
+    if (vice || vacancy) {
+      return `${vice || ''}${(vice && vice !== 'Multiple Incumbents') ? ', ' : ''} ${vacancy || ''}`;
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaLeg/AgendaLeg.jsx:252]: Do we want to do a check for vacancy here as well? to avoid `Smith, John,`. Not a big deal, so up to you

File: src/sass/_agendaItemLegsForm.scss [hunk 6]
```diff
@@ -36,7 +36,7 @@
       min-height: 105px;
     }
 
-    $cols-rows: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 ];
+    $cols-rows: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13 ];
```
[the remark this answers is not in the document]
Comment (M00staff) [reply src/sass/_agendaItemLegsForm.scss:39]: I think it looks better without the minimum height?


image

image

File: src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx (hunk 1 above)
Comment (M00staff) [reply src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx:127]: nah - i think the codebase need more comments, if anything

the 'why' for certain bits like this could be confusing to new folks, imo 🤷

File: src/Components/Agenda/AgendaLeg/AgendaLeg.jsx (hunk 5 above)
Comment (M00staff) [reply src/Components/Agenda/AgendaLeg/AgendaLeg.jsx]: yea - this makes sense, especially considering i dont know what the real data will look like and what all comes back null

updated!

File: src/sass/_agendaItemLegsForm.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply src/sass/_agendaItemLegsForm.scss:39]: Ah, i think my comment is outdated - now that Vice is at the end, we don't need to update those variables, but before when you had it somewhere in the middle, those cells' stylings got shifted

**Review by elizabeth-jimenez — APPROVED** [review]

File: src/sass/_agendaItemLegsForm.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (M00staff) [reply src/sass/_agendaItemLegsForm.scss:39]: ahhh - makes sense!
